src/Parsers/ParserInsertQuery.cpp

```diff
@@ -133,6 +133,20 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     Pos before_values = pos;
     String format_str;
 
+    /// Read SETTINGS if they are defined
+    if (s_settings.ignore(pos, expected))
+    {
+        /// Settings are written like SET query, so parse them with ParserSetQuery
+        ParserSetQuery parser_settings(true);
+        if (!parser_settings.parse(pos, settings_ast, expected))
+            return false;
+
+        /// We should move data pointer after all settings, in case of:
+        /// - INSERT SELECT
+        /// - INSERT WATCH
+        before_values = pos;
+    }
+
```

Comment:
We can write `Pos before_values = pos;` here to assign `before_values` in only one place

Reply:
Right, thanks.